Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/PreserveOutputsTests.cs

```diff
@@ -275,8 +310,65 @@ public void PreserveOutputsOpaqueTest()
             // No cache hit
             outputContents = RunSchedulerAndGetOutputContents(output, cacheHitAssert: false, id: pipA.Process.PipId);
             
-            // Preserve outputs is not supported for opaque dirs
+            // As the opaque output is preserved, the pip appended the existing file.
+            XAssert.AreEqual(CONTENT_TWICE, outputContents);
+
+            outputContents = RunSchedulerAndGetOutputContents(output, cacheHitAssert: true, id: pipA.Process.PipId);
+
+            // Cache hit and the appended file (CONTENT_TWICE) should remain the same.
+            XAssert.AreEqual(CONTENT_TWICE, outputContents);

```

Comment:
This is a breaking change. We should do a couple things
1. Make sure to include it in our release notes next week
2. Coordinate with Office to make sure they won't be blocked from taking the next release

Reply:
I chatted with Ben this morning. He is aware of the change and he is fine with it. After we have the RC, he will get the changes done on the office side.

---
In reply to: [287426249](https://github.com/microsoft/BuildXL/pull/369#discussion_r287426249) [](ancestors = 287426249)

Reply:
I doubt it will matter to them but could double check on APEX and WDG as well?

---
In reply to: [287525535](https://github.com/microsoft/BuildXL/pull/369#discussion_r287525535) [](ancestors = 287525535,287426249)